lektor_markdown_header_anchors.py

```diff
@@ -14,32 +22,22 @@ class MarkdownHeaderAnchorsPlugin(Plugin):
     def on_markdown_config(self, config, **extra):
         class HeaderAnchorMixin(object):
             def header(renderer, text, level, raw):
-                anchor = slugify(raw)
-                renderer.meta['toc'].append((level, anchor, Markup(text)))
+                anchor = uniquify(renderer.meta['uniquify'], raw)
+                renderer.meta['toc'].append(TocEntry(level, anchor, Markup(text), []))
                 return '<h%d id="%s">%s</h%d>' % (level, anchor, text, level)
         config.renderer_mixins.append(HeaderAnchorMixin)
 
     def on_markdown_meta_init(self, meta, **extra):
         meta['toc'] = []
+        meta['uniquify'] = Counter()
```

Comment:
Why do we need this? not following :-p

Reply:
A counter to count the number of times each anchor appears. Then append ~2, ~3, etc, to ensure the same anchor does not appear more than once.

Reply:
Originally,  while parsing, the plugin would create a tuple of (level, anchor, markup) and then during toc tree creation, it would create a TocEntry. I moved the creation of the TocEntry, which required adding level field there. 

It simplifies the code a bit.

Reply:
No I  understand, I was not sure where it was being used. But now I realized it :-)